src/Interpreters/Set.h

```diff
@@ -217,6 +217,8 @@ class MergeTreeSetIndex
     BoolMask checkInRange(const std::vector<Range> & key_ranges, const DataTypes & data_types) const;
 
 private:
+    // If all arguments in tuple are key columns, we can optimize NOT IN when there is only one element.
+    bool all_key;
```

Comment:
I think it's better to name this variable something like `maps_all_keys`

Reply:
hmm, but why maps? How about `has_all_keys`?

Reply:
That sounds good too.